Docs: New Getting Started Page

dericksozo opened the pull request:
(docs): Adding the first draft of the new Getting Started page along with the new structure (draft).

dericksozo commented:
Hey @PazYanoverr, this PR introduces a new Getting Started page draft that reorganizes our documentation structure. The proposed changes include:

- New overview page highlighting key platform engineering capabilities
- Restructured "Getting Started" sidebar section that will feature quickstart (first service creation), dedicated example services section, and In-depth Application introduction serving as documentation table of contents/roadmap

What are your initial thoughts on this first draft and this direction for the getting start sidebar re-structuring?

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
 [Create your first service](/first-service/) with Amplication's AI, Jovu, and generate a .NET or Node.js _production-ready codebase in minutes_.
 
-## Customize Your Generated Service
+## AI-powered Backend Generation
```
Since our advantages reside mostly in the standards and tech debts, I think we should push the "AI-powered" section to be after the 'tech debt" section.
WDYT?

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
+## Embed Your Organization's Best Practices
 
-## Key Features
+Define and maintain standardized service architectures through live templates and private plugins.
```
It's not only about the architecture- but also about code conventions, tools, etc. I think that mentioning architecture here is a little confusing

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
 
-## Key Features
+Define and maintain standardized service architectures through live templates and private plugins.
+Create template with pre-configured settings, code generator versions, and plugins.
```
```suggestion
Create template with pre-configured settings, code generator version, and plugins.
```

PazYanoverr replied:
In addition- I think we need to say something about the live templates and plugins- that any update of these will be reflected as a PR in the services- so the best practices and standards are evolving or such.

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
-- **[Multiple Databases](/getting-started/#databases)**: Support for PostgreSQL, MySQL, MongoDB, MS SQL Server, and more.
-- **Deployment & CI/CD**: Simplify and automate your CI/CD pipeline with our plugins and streamline deployments to popular options like [AWS ECS](/deploy/aws/ecs), [Docker Desktop](/deploy/docker-desktop/), and [Kubernetes](/deploy/kubernetes).
+- **Live Service Templates**
+- **[Private Plugins](/getting-started/plugins/)**: Extend functionality with private plugins or use our community plugins.
```
The link here is wrong- in case the text is "private plugin" - the link should be to the private plugin page (and then the text after this "title" should be updated. Otherwise- we should call it plugins (and maybe add a link to the private plugins for the "private plugins" text.

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
-- Explore our [sample applications](https://github.com/amplication/sample-app) on GitHub.
+- Technical Debt Management
+- Service Catalog
+- **[Seamless Git Integration](/smart-git-sync/)**: Smart git sync with GitHub, Bitbucket, GitLab, or AWS CodeCommit.
```
Not sure this is relevant here (unless rephrase it that all the service updates due to technical debts handling- is available in Git or such

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
+- Technical Debt Management
+- Service Catalog
+- **[Seamless Git Integration](/smart-git-sync/)**: Smart git sync with GitHub, Bitbucket, GitLab, or AWS CodeCommit.
+- **[Break Your Monolith](/concepts/event-driven-architecture/)**: Create new services with a modern architecture or modernize your existing apps into a event-driven microservices architecture.
```
Not relevant here. I would omit it

PazYanoverr commented on docs/welcome.md:
```diff
@@ -1,54 +1,59 @@
-- Check out our [YouTube channel](https://www.youtube.com/c/Amplicationcom) for video tutorials.
-- Read our [blog](https://amplication.com/blog) for in-depth articles and updates.
+Enable developers to quickly create standardized services through our Service Catalog.
+Import existing databases, extend functionality through custom code, leverage multiple database options, and create custom plugins.
```
1. Let's push the import to the end of the list here (and below)
2. Let's remove the "extend functionality through custom code" - here and below (as it adds nothing "real" to our capabilities list).
3. Instead of talking about creating custom plugins- let's talk about using plugins to add functionality to the services efficiently and in a standard way. This can be done using the community plugins or creating private plugins. We can mention here the categories of plugins like deployment, security etc.

PazYanoverr commented:
@dericksozo - The "overview" page looks good! I added a few comments.
Regarding the sidebar - I wish it reflects quickly Amplication major capabilities and advantages. A roadmap and examples are nice, but I think we need to "get to the point" much faster here. I.e. have here pages mostly about our live templates, technical debt handling, and - maybe- plugins + private plugins. I believe these pages can also appear as part of the "regular" menu, but these issues should be emphasized in the documentation - i.e. in the sidebar menu, in the top menu etc.
WDTY?

dericksozo commented:
Hello @PazYanoverr,

I implemented the feedback into the Getting Started page. I added a new "Core Platform Features" section to the sidebar, showcasing Platform Console and Private Plugins pages. Once Live Templates and Service Catalog PRs are merged, they will be added to this section. We should also consider creating a dedicated Technical Debt Management page to complete the core feature documentation.

Thoughts?

PazYanoverr commented:
@dericksozo - Sounds good. In addition- I fully agree regarding the Technical Debt Management page that should be created. Let's add it as a new task.
One small comment: I think that the Documentation title should "hold" all the pages, not only the two pages of Overview and "Create your first..." -
![image](https://github.com/user-attachments/assets/f8ce040e-4385-42ce-a836-19092293fb83)
